infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationClientRegisterListener.java

```diff
@@ -120,11 +122,17 @@ private void handleRegistrationTableEntries(List<CorfuStreamEntry> registrationT
             if (entry.getOperation().equals(CorfuStreamEntry.OperationType.UPDATE)) {
                 String clientName = ((ClientRegistrationId) entry.getKey()).getClientName();
                 ReplicationModel model = ((ClientRegistrationInfo) entry.getPayload()).getModel();
-                // TODO (V2): Currently we add a default subscriber for logical group use case instead of listening
-                //  on client registration. Subscriber should be added upon registration after grpc stream for session
-                //  creation is added.
-                // configManager.onNewClientRegister(subscriber);
-                // sessionManager.createOutgoingSessionsBySubscriber(subscriber);
+                LogReplication.ReplicationSubscriber subscriber = LogReplication.ReplicationSubscriber.newBuilder()
+                        .setClientName(clientName).setModel(model).build();
+                if (model.equals(ReplicationModel.FULL_TABLE)
+                        || model.equals(ReplicationModel.LOGICAL_GROUPS) || model.equals(ReplicationModel.ROUTING_QUEUES)) {
+                    if(sessionManager.getReplicationContext().getIsLeader().get()) {
+                        configManager.onNewClientRegister(subscriber);
+                        Set<LogReplication.ReplicationSubscriber> subscriberSet = ConcurrentHashMap.newKeySet();
+                        subscriberSet.add(subscriber);
+                        sessionManager.createSessions(subscriberSet, false);
```

Comment:
why do we need to pass a set for a single subscriber?  Can we avoid overloading the current 'createSessions' method?